Grow.c

```diff
@@ -3068,6 +3069,39 @@ static void catch_term(int sig)
 	sigterm = 1;
 }
 
+/**
+ * handle_forking() - Handle forking, helper function for reshapes.
+ *
+ * @forked: is process forked.
+ * @devname: device name.
+ * @service: wheather a service should be started or just fork.
+ *
+ * Returns: -1 if failed to fork,
+ *           0 if currently child process,
+ *           1 process forked, passed to service or delegated to udev.
+ */
+static int handle_forking(bool forked, char *devname, bool service)
+{
+	/* Nothing to do if already forked. */
+	if (forked)
+		return 0;
+
+	/* Just fork if no service. */
+	if (service) {
+		/* Allow udev to trigger service after switchroot. */
+		if (in_initrd())
+			return 1;
+		if (continue_via_systemd(devname, GROW_SERVICE, NULL))
+			return 1;
+	}
+
+	switch (fork()) {
```

Comment:
why not just `return fork()`

Reply:
Just for unification reasons, so we don't return 1s or PIDs depending what path is taken.